### ISSUE TITLE
Endless method with no args aren't parsed with `ruby_parser` `3.18.0`

### ISSUE THREAD

**Issue by dgholz** [presidentbeef/brakeman#1652]
### Background

Brakeman version: 5.1.2
Rails version: 6.1.4.1
Ruby version: 3.0.2

Link to Rails application code: https://github.com/dgholz/reviewdog-brakeman/pull/5/files (an example project)

### Issue

I have some code in a model that looks like:
```
  def self.foo = 'bar'
```
Brakeman fails to parse it correctly and errors out. I see `ruby_parser` was bumped for #1593 but there was a follow-up fix for endless methods in https://github.com/seattlerb/ruby_parser/commit/c0526aa8cde85032920d36fe257fc9c96037d835 that got released in `3.18.1`. I manually updated the gems bundled with Brakeman to use that version, and it runs on my code just fine.

#### Other Error

I ran Brakeman with `--debug` in the above branch (https://github.com/dgholz/reviewdog-brakeman/tree/parse_error)

Stack trace:

```
$ bundle exec brakeman --debug
[Notice] Using configuration in /Users/danielholz/code/reviewdog-brakeman/config/brakeman.yml
Loading scanner...
Processing application in /Users/danielholz/code/reviewdog-brakeman
Processing gems...                    
Parsing Gemfile
Processing configuration...           
Parsing config/environment.rb
Parsing config/application.rb
Parsing config/environments/production.rb
[Notice] Escaping HTML by default
Parsing files...                      
Parsing app/channels/application_cable/channel.rb
Parsing app/controllers/application_controller.rb
Parsing app/helpers/application_helper.rb
Parsing app/channels/application_cable/connection.rb
Parsing app/controllers/articles_controller.rb
Parsing app/helpers/articles_helper.rb
Parsing app/helpers/comments_helper.rb
Parsing app/controllers/comments_controller.rb
Parsing app/jobs/application_job.rb
Parsing app/mailers/application_mailer.rb
Parsing app/models/application_record.rb
Parsing app/models/article.rb
Parsing config/application.rb
Parsing app/models/comment.rb
Parsing config/boot.rb
Parsing config/environment.rb
Parsing config/environments/development.rb
Parsing config/initializers/backtrace_silencers.rb
Parsing config/initializers/assets.rb
Parsing config/environments/test.rb
Parsing config/initializers/application_controller_renderer.rb
Parsing config/environments/production.rb
Parsing config/initializers/content_security_policy.rb
Parsing config/initializers/cookies_serializer.rb
Parsing config/initializers/filter_parameter_logging.rb
Parsing config/initializers/mime_types.rb
Parsing config/initializers/inflections.rb
Parsing config/initializers/permissions_policy.rb
Parsing config/initializers/wrap_parameters.rb
Parsing config/puma.rb
Parsing config/routes.rb
Parsing config/spring.rb
Parsing /Users/danielholz/code/reviewdog-brakeman/app/views/articles/index.html.erb
Parsing app/views/articles/index.html.erb
Parsing /Users/danielholz/code/reviewdog-brakeman/app/views/layouts/application.html.erb
Parsing app/views/layouts/application.html.erb
Parsing /Users/danielholz/code/reviewdog-brakeman/app/views/layouts/mailer.html.erb
Parsing app/views/layouts/mailer.html.erb
Parsing /Users/danielholz/code/reviewdog-brakeman/node_modules/turbolinks/src/turbolinks/BANNER.js.erb
Parsing node_modules/turbolinks/src/turbolinks/BANNER.js.erb
Detecting file types...               
Processing initializers...            
Processing /Users/danielholz/code/reviewdog-brakeman/config/initializers/assets.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/initializers/backtrace_silencers.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/initializers/cookies_serializer.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/initializers/filter_parameter_logging.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/initializers/wrap_parameters.rb
Processing libs...                    
Processing /Users/danielholz/code/reviewdog-brakeman/app/channels/application_cable/channel.rb
Processing /Users/danielholz/code/reviewdog-brakeman/app/channels/application_cable/connection.rb
Processing /Users/danielholz/code/reviewdog-brakeman/app/helpers/application_helper.rb
Processing /Users/danielholz/code/reviewdog-brakeman/app/helpers/articles_helper.rb
Processing /Users/danielholz/code/reviewdog-brakeman/app/helpers/comments_helper.rb
Processing /Users/danielholz/code/reviewdog-brakeman/app/jobs/application_job.rb
Processing /Users/danielholz/code/reviewdog-brakeman/app/mailers/application_mailer.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/boot.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/environment.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/environments/development.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/environments/test.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/puma.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/routes.rb
Processing /Users/danielholz/code/reviewdog-brakeman/config/spring.rb
Processing routes...                  
Parsing config/routes.rb
Processing templates...               
Processing /Users/danielholz/code/reviewdog-brakeman/app/views/articles/index.html.erb
Processing /Users/danielholz/code/reviewdog-brakeman/app/views/layouts/application.html.erb
Processing /Users/danielholz/code/reviewdog-brakeman/app/views/layouts/mailer.html.erb
Processing /Users/danielholz/code/reviewdog-brakeman/node_modules/turbolinks/src/turbolinks/BANNER.js.erb
Processing data flow in templates...  
Processing articles/index
Processing layouts/application
Processing layouts/mailer
Processing node_modules/turbolinks/src/turbolinks/BANNER
Processing models...                  
Processing /Users/danielholz/code/reviewdog-brakeman/app/models/application_record.rb
Processing /Users/danielholz/code/reviewdog-brakeman/app/models/article.rb
Processing /Users/danielholz/code/reviewdog-brakeman/app/models/comment.rb
bundler: failed to load command: brakeman (/Users/danielholz/.gem/ruby/3.0.2/bin/brakeman)
/Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/tracker/method_info.rb:36:in `very_simple_method?': undefined method `length' for nil:NilClass (NoMethodError)
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/tracker/collection.rb:125:in `add_simple_method_maybe'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/tracker/collection.rb:53:in `add_method'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/processors/lib/module_helper.rb:88:in `process_defs'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/processors/lib/processor_helper.rb:13:in `block in process_all!'
        from (eval):3:in `map!'
        from (eval):3:in `map!'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/processors/lib/processor_helper.rb:11:in `process_all!'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/processors/lib/module_helper.rb:54:in `handle_class'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/processors/model_processor.rb:35:in `process_class'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/ruby_parser/bm_sexp_processor.rb:76:in `block in process'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/ruby_parser/bm_sexp_processor.rb:113:in `in_context'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/ruby_parser/bm_sexp_processor.rb:72:in `process'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/processors/model_processor.rb:21:in `process_model'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/processor.rb:55:in `process_model'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/scanner.rb:331:in `process_model'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/scanner.rb:326:in `block in process_models'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/scanner.rb:340:in `block in track_progress'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/scanner.rb:337:in `each'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/scanner.rb:337:in `track_progress'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/scanner.rb:324:in `process_models'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/scanner.rb:63:in `process'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman.rb:392:in `scan'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman.rb:86:in `run'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/commandline.rb:157:in `run_brakeman'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/commandline.rb:125:in `regular_report'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/commandline.rb:166:in `run_report'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/commandline.rb:35:in `run'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/lib/brakeman/commandline.rb:20:in `start'
        from /Users/danielholz/.gem/ruby/3.0.2/gems/brakeman-5.1.2/bin/brakeman:10:in `<top (required)>'
        from /Users/danielholz/.gem/ruby/3.0.2/bin/brakeman:23:in `load'
        from /Users/danielholz/.gem/ruby/3.0.2/bin/brakeman:23:in `<top (required)>'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:63:in `load'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:63:in `kernel_load'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:28:in `run'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/cli.rb:474:in `exec'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/cli.rb:30:in `dispatch'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/cli.rb:24:in `start'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.2.22/libexec/bundle:49:in `block in <top (required)>'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/3.0.0/bundler/friendly_errors.rb:128:in `with_friendly_errors'
        from /Users/danielholz/.rubies/ruby-3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.2.22/libexec/bundle:37:in `<top (required)>'
        from /Users/danielholz/.rubies/ruby-3.0.2/bin/bundle:23:in `load'
        from /Users/danielholz/.rubies/ruby-3.0.2/bin/bundle:23:in `<main>'
```

**Comment by presidentbeef** [comment]
Hi @dgholz Until the next Brakeman release, please try the `brakeman-lib` gem so it will pick up your version of `ruby_parser`. The `brakeman` gem vendors all its dependencies.

**Comment by presidentbeef** [comment]
This should be resolved now with Brakeman 5.2.0.
